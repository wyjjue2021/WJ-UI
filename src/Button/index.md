---
title: Button 按钮
nav:
  title: 组件
  order: 2
group:
  title: 通用
  order: 1
---

# Button 按钮

按钮，用于开始一个即时操作。

## 代码演示

### 基本用法

<code src="./demo/index.tsx"></code>

<API></API>