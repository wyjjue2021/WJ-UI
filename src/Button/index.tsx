import React from 'react'
import t from 'prop-types'

export interface ButtonProps {
    /**
   * 可以这样写属性描述
   * @description       设置按钮类型	
   * @description.zh-CN 还支持不同的 locale 后缀来实现多语言描述
   * @default           支持定义默认值
   */
    type?: 'primary' | 'dashed' | 'text' | 'link'
}



const prefixCls = 'wj-button';

const Button:React.FC<ButtonProps> = ({children, type = 'primary', ...rest }) => {
    return <div className={prefixCls}
        style = {{}}>
        {children}
    </div>
}


Button.propTypes = {
    type: t.oneOf(['primary', 'dashed', 'text', 'link']),
  };

export default Button;
