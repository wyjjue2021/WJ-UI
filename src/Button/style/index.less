@popupPrefix: wj-button;

.@{popupPrefix} {
  padding: 20px;
  background: white;
  border-radius: 3px;
  color: white;
}