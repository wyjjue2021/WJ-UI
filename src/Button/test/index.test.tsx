import React from 'react';
import { render } from '@testing-library/react';
import Button from '../index';

describe('<Button />', () => {
  test('should render default', () => {
    const { container } = render(<Button>default</Button>);
    expect(container).toMatchSnapshot();
  });
});