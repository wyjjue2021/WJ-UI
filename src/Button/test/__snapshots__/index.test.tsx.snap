// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> should render default 1`] = `
<div>
  <div
    class="wj-button"
  >
    default
  </div>
</div>
`;
